sys/net/link_layer/Kconfig.lorawan

```diff
@@ -0,0 +1,483 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+config HAVE_LORAWAN
+    bool
+    help
+        Indicates that LoRaWAN headers are present.
+
+menuconfig KCONFIG_USEMODULE_LORAWAN
+    bool "Configure LoRaWAN MAC"
+    depends on USEMODULE_GNRC_LORAWAN || USEPKG_SEMTECH_LORAMAC || USEMODULE_RN2XX3
+    help
+        Configure LoRaWAN module using Kconfig. This may override configurations
+        in LoRa PHY.
+
+if KCONFIG_USEMODULE_LORAWAN
+
+choice
+    bool "LoRaWAN region"
+    depends on USEMODULE_GNRC_LORAWAN
+    default LORAMAC_REGION_EU_868
+    help
+        Configure the LoRaWAN region.
+
+config LORAMAC_REGION_EU_868
+    bool "EU863-870"
+
+config LORAMAC_REGION_IN_865
+    bool "IN865-867"
+
+endchoice
+
+choice
+    bool "Join procedure"
+    depends on USEMODULE_GNRC_LORAWAN
+    default LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    help
+        Configure the join procedure for end node activation. There are two
+        options, Over The Air Activation (OTAA) results in device sending
+        join request to the network whereas in the case of Activation By
+        Personalization (ABP) the user enters the activation keys manually. OTAA
+        is the preferred and most secure way to connect to a LoRaWAN network.
+        For quick testing ABP is preferred as the device can transmit right
+        away without waiting for the network to provision the keys.
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+    bool "Over the air activation (OTAA)"
+
+config LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP
+    bool "Activation by personalization (ABP)"
+
+endchoice
+
+if LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA || USEPKG_SEMTECH_LORAMAC
+
+config LORAMAC_DEV_EUI_DEFAULT
+    string "Device EUI"
+    default "0000000000000000"
+    help
+        Configure the default device EUI for join procedure.
+
+config LORAMAC_APP_EUI_DEFAULT
+    string "Application EUI"
+    default "0000000000000000"
+    help
+        Configure the default application EUI for join procedure.
+
+config LORAMAC_APP_KEY_DEFAULT
+    string "Application key"
+    default "00000000000000000000000000000000"
+    help
+        Configure the default application key for join procedure.
+
+endif # LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA
+
+if LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP && USEMODULE_GNRC_LORAWAN
```

Comment:
Not possible to configure ABP credentials for other stacks??

Reply:
Yes. Fixups on the way :)

Reply:
I shall extend the rest of the configurations in another PR a major rework is being planned.